### DISCUSSION TITLE
Are paired shortcodes supported in Quarto?

### DISCUSSION THREAD

**Discussion by sidneygijzen** [quarto-dev/quarto-cli#10855] [Q&A] [answered]
hi,

I've been reading the shortcodes docs page, but unfortunately couldn't find if paired (combined with nested) shortcodes are supported by Quarto. I'm looking for something similar to what's possible [in Hugo](https://gohugo.io/templates/shortcode/#nested-shortcode-image-gallery).

In my use-case, which is a website with a custom design, I would like to do something like:

```md
{{< section-image-bg image="path/to/img" >}}
  ## a header

  The content for this section
  
  {{< button-primary label="the button text" >}}
{{< /section-image-bg >}}
```

In other words: is it possible to use a Quarto shortcode as a paired shortcode?

**Comment by cscheid** [comment, marked as answer]
No - Quarto uses divs, spans and Lua filters instead.

**Reply by cscheid** [reply]
In other words, you'd write something like

````
---
title: my document
filters:
  - my_filter.lua
---

::: {.section-image-bg image="path/to/img"}

## a header

The content for this section
  
{{< button-primary label="the button text" >}}
:::
````

And then a Lua filter that catches divs with the class `.section-image-bg` and processes it as you see fit:

```
-- my_filter.lua
function Div(div)
  if div.classes:includes("section-image-bg") then
    ...
  end
end
```

**Reply by sidneygijzen** [reply]
OK. Thank you for the fast response and the example code.
